webui/react/src/hooks/useModal/Checkpoint/useModalCheckpointRegister.tsx

```diff
@@ -283,30 +276,26 @@ const useRegisterCheckpointModal = (onClose?: (checkpointUuid?: string) => void)
     };
 
     return modalProps;
-  }, [ generateModalContent, handleCancel, handleOk ]);
+  }, [ getModalContent, handleCancel, handleOk ]);
 
-  // Detect modal state change and update.
   useEffect(() => {
-    if (!modalState.visible) return;
+    fetchModels();
+  }, [ fetchModels ]);
 
-    const modalProps = generateModalProps(modalState);
-    if (modalRef.current) {
-      modalRef.current.update(prev => ({ ...prev, ...modalProps }));
-    } else {
-      modalRef.current = Modal.confirm(modalProps);
-    }
-  }, [ generateModalProps, modalState ]);
+  useEffect(() => {
+    return () => canceler.abort();
+  }, [ canceler ]);
 
-  // When the component using the hook unmounts, remove the modal automatically.
+  /**
+   * When modal props changes are detected, such as modal content
+   * title, and buttons, update the modal.
+   */
   useEffect(() => {
-    return () => {
```

Comment:
not needed anymore with the context shift? or is this to change the behavior

Reply:
now that it is composing from `useModal`, this behavior is inherited from there.